### PR TITLE
Add Copy Path to editor tab context menu

### LINKED ISSUE #7289: Copy script path from tabs
Many IDEs allow you to copy the name of a script or the full path of a script by right-clicking on the tab (open script name) and selecting 'copy...' from a dropdown menu.  It would be very useful to have something similar, as I am often wanting to send paths to a file to colleagues.

### PR DESCRIPTION
**Pull request by gtritchie**
### Intent

- Fixes #7289
- IDE's typically have a way to copy the path of an open file via context menu, now we do, too
screenshot of document tab context menu showing new Copy Path command
- Also added to command palette as "Copy Document Path" for keyboard-only use

### Approach

- Went for simplest approach, just copy the entire path, including filename, to the clipboard
- On Windows Desktop, a leading `~` will be resolved to a full path, since this isn't a native concept on Windows, but on Server, other desktop platforms, and RDP remote sessions, the `~` will be left as-is

### QA Notes

- Test in source columns and in popped-out satellite source editors using both Command Palette and tab context menu
- Test in a variety of browsers; the code I'm using for copying came from [stackoverflow](https://stackoverflow.com/questions/400212/how-do-i-copy-to-the-clipboard-in-javascript) and seems to cover all major browsers we support in my testing, but it's possible we might find some edge cases (if so, we could consider using [clipboard.js](https://clipboardjs.com/), instead)
- I expect people will request variations on this, e.g. just copy the filename, just copy the path without the filename, resolve the entire path on all platforms, etc., but just getting in the basic functionality for v1.4-juliet-rose

### CODE REVIEW

File: src/gwt/src/org/rstudio/studio/client/workbench/views/files/model/FilesServerOperations.java [hunk 1]
```diff
@@ -98,4 +98,9 @@ void writeConfigJSON(String relativePath,
    void readConfigJSON(String relativePath,
                  boolean logErrorIfNotFound,
                  ServerRequestCallback<JavaScriptObject> requestCallback);
+
+   // Use VERY sparingly; we generally don't want to expose
+   // non-aliased paths to other parts of the client codebase
+   // (most client-side APIs assume paths are aliased)
+   String resolveAliasedPath(FileSystemItem file);
```
Comment (jmcphers) [inline src/gwt/src/org/rstudio/studio/client/workbench/views/files/model/FilesServerOperations.java:105]: Nit: If you give this a doxygen-style comment (i.e. `/**`) it will show up in IDEs to help discourage use during autocomplete.

**Review by jmcphers — APPROVED** [review]
LGTM, this is a great improvement!

File: src/gwt/src/org/rstudio/studio/client/workbench/views/files/model/FilesServerOperations.java (hunk 1 above)
Comment (gtritchie) [reply src/gwt/src/org/rstudio/studio/client/workbench/views/files/model/FilesServerOperations.java:107]: Done!
